.github/workflows/docker.yml

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2024 Hemi Labs, Inc.
+# Use of this source code is governed by the MIT License,
+# which can be found in the LICENSE file.
+
+# GitHub Actions workflow to publish Docker images.
+name: "Docker"
+on:
+  push:
+
+concurrency:
+  group: "docker-${{ github.workflow }}-${{ github.event.number || github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  # Build and publish Docker image
+  docker:
+    name: "Docker"
+    if: ${{ github.repository_owner == 'hemilabs' }}
+    runs-on: "ubuntu-latest"
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+
+      - name: "Setup QEMU"
+        uses: docker/setup-qemu-action@v3
+
+      - name: "Setup Docker Buildx"
+        uses: docker/setup-buildx-action@v3
+
+      - name: "Login to DockerHub"
+        uses: docker/login-action@v3
+        with:
+          username: "${{ secrets.DOCKERHUB_USERNAME }}"
+          password: "${{ secrets.DOCKERHUB_TOKEN }}"
+
+      - name: "Login to GitHub Container Registry"
+        uses: docker/login-action@v3
+        with:
+          registry: "ghcr.io"
+          username: "${{ github.repository_owner }}"
+          password: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: "Prepare"
+        id: "prepare"
+        run: |
+          echo "sha_short=$(git rev-parse --short HEAD)" >> "$GITHUB_OUTPUT"
+          echo "build_date=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> "$GITHUB_OUTPUT"
+
+      - name: "Build and push"
+        uses: docker/build-push-action@v5
+        with:
+          context: "${{ github.workspace }}"
+          platforms: "amd64"
+          file: "${{ github.workspace }}/op-node/Dockerfile"
+          push: true
+          build-args: |
+            VERSION=${{ steps.prepare.outputs.sha_short }}
+            VCS_REF=${{ github.sha }}
+            BUILD_DATE=${{ steps.prepare.outputs.build_date }}
```

Comment:
I don't see these used in the Dockerfile, could we add labels to the Dockerfile (see heminetwork Dockerfiles)?

Reply:
sure 👍